zmq/tests/test_socket.py

```diff
@@ -396,7 +397,19 @@ def test_ipc_path_max_length_msg(self):
             s.bind('ipc://{0}'.format('a' * (zmq.IPC_PATH_MAX_LEN + 1)))
         except zmq.ZMQError as e:
             self.assertTrue(str(zmq.IPC_PATH_MAX_LEN) in e.strerror)
-    
+
+    @mark.skipif(windows, reason="ipc not supported on Windows.")
+    def test_ipc_path_no_such_file_or_directory_message(self):
+        """Display the ipc path in case of an ENOENT exception"""
+        s = self.context.socket(zmq.PUB)
+        self.sockets.append(s)
+        invalid_path = '/foo/bar'
+        with pytest.raises(zmq.ZMQError) as error:
+            s.bind('ipc://{0}'.format(invalid_path))
+        error_message = str(error.value)
+        assert invalid_path in error_message
+        assert "no such file or directory" in error_message.lower()
```

Comment:
Can you add one more assert that `error.errno == ENOENT`?

Reply:
Thanks for the review!

I pushed a fixup with that change.

Tell me if it is good for you and I will squash it. Or feel free to squash it yourself before merging. :+1:

Reply:
no need for that. Thanks!